### ISSUE TITLE
[Security Review] Pixie

### ISSUE THREAD

**Issue by aimichelle** [cncf/tag-security#1068]
Hi all! We recently applied [Pixie](https://github.com/pixie-io/pixie) for incubation. Since Pixie leverages eBPF and requires privileged access in users' clusters, we thought conducting a security review would be helpful and reassuring for our users.
We've already gone ahead and drafted a self-assessment. We can submit the self-assessment in a PR if that is the next best step!

Project Name: Pixie

Github URL: https://github.com/pixie-io/pixie

CNCF project stage and issue (NA if not applicable): Sandbox, [applying for incubation](https://github.com/cncf/toc/pull/1001)

Security Provider: no

- [x] Identify team
   - [X] Project security lead @aimichelle 
   - [X] Lead security reviewer @JustinCappos 
   - [X] 1 or more additional reviewer(s): @ragashreeshekar @JonZeolla @victorjunlu  Observers: @krishnakv 
   - [X] Every reviewer has read [security reviewer guidelines](https://github.com/cncf/tag-security/blob/main/assessments/guide/security-reviewer.md) and stated declaration of conflict
   - [x] Sign off by 2 chairs on reviewer conflicts
- [x] Create slack channel (e.g. #sec-assess-projectname)
- [x] Project lead provides [draft document](https://docs.google.com/document/d/1H4C9uGVxLsLZijxIem_vwhOWLupBqtlwMduI2b3UL8M/edit#heading=h.arjzsoy5mukn)
- [x] "Naive question phase" Lead Security Reviewer asks clarifying questions
- [x] Assign issue to security reviewers
- [x] Initial review
- [x] Presentation & discussion
- [x] Share draft findings with project
- [x] Assessment summary and doc checked into /assessments/projects/project-name (require at least 1 co-chair approval)
- [ ] CNCF TOC presentation (if requested by TOC)

**Comment by JustinCappos** [comment]
I'm happy to lead the security review here.  I'd like to ask others from TAG-Security to please volunteer as additional assessors.

I have no hard or soft conflicts with the work.

**Comment by ragashreeshekar** [comment]
I'd like to join as an assessor. 
No conflicts from my end as well.

**Comment by aimichelle** [comment]
Thank you @JustinCappos and @ragashreeshekar ! I will be the representative from the Pixie side.
Let me know what the next steps are and if there's anything I can do to help.

Here is a draft of our self-assessment if it helps: https://docs.google.com/document/d/1H4C9uGVxLsLZijxIem_vwhOWLupBqtlwMduI2b3UL8M/edit#heading=h.arjzsoy5mukn

**Comment by JustinCappos** [comment]
Okay, I've added some comments to the self-assessment as part of the "naive questions phase".  I also created the sec-assess-pixie channel on the CNCF slack, if you'd like to join, @aimichelle .

**Comment by JonZeolla** [comment]
@JustinCappos I'd like to join the team as a security reviewer.  I also have no conflicts to disclose

**Comment by lumjjb** [comment]
Would this be a good candidate to try out the lightweight threat modeling? @JustinCappos @ragashreeshekar @sublimino

**Comment by sublimino** [comment]
@lumjjb yes, I think it would suit it well.  

We identified during the introductory call that there's a large surface area, and it's important to keep scope well-defined and tight. It may take a couple of sessions rather than one.

**Comment by JustinCappos** [comment]
I'm not as sure.  I mean, they did all the prep work, so why give them a cursory process?

**Comment by sublimino** [comment]
Apologies for the confusion — happy to defer to your lead @JustinCappos.

I mistook this for Flatcar with my initial answer and am not on this team :pray:

**Comment by victorjunlu** [comment]
I am interested in joining this assessment.

**Comment by JonZeolla** [comment]
@JustinCappos I still have no conflicts to disclose

**Comment by JustinCappos** [comment]
@lumjjb @achetal01 @sublimino Can we get signoffs from (at least) two of you on the conflict statements please?

**Comment by krishnakv** [comment]
@JustinCappos , I would like to volunteer for this review. Since this is my first review, happy to participate as an observer.

Have gone through the assessment guide and I have no hard or soft conflicts to report for this review.

**Comment by JonZeolla** [comment]
@PushkarJ wanted to bring your attention to the [above comment](https://github.com/cncf/tag-security/issues/1068#issuecomment-1583914630) in case you could help unblock this.  Thanks!

**Comment by JonZeolla** [comment]
I did some poking around on the lightweight threat modeling and I think [this is the template we'd start with](https://docs.google.com/document/d/1tuGtKrjcreDFlHcXYCTjLvy3mjyamdQzwCZr6uqFcR4/edit), in case others also wanted to take a look

**Comment by JustinCappos** [comment]
Okay, I've just checked the box for the chair sign-off.  If they don't like it, they can complain later, but I'm tired of this delaying things.

This means we start the naive questions phase.  We've already done some of this process, but I'll make a quick pass again today.  

After this all the reviewers will get looped in.  Since many of us have already been looking over things, I'd expect this to go quite quickly.

**Comment by JustinCappos** [comment]
I've completed my "Naïve questions pass" over the document.  I think it looks very good!

@aimichelle , I'd suggest you integrate the answers to the comments into the text.  This way we can resolve those comment threads and you can have a more complete and consistent document that is easier to read.  Please ping us when this is ready.

Once this is done, then I'll open this to all reviewers and we will go over it again.  As I said in my prior comment, this should be a quick process.

**Comment by achetal01** [comment]
Justin 
I agree, Some very good comments by the reviewers. If we can get the responses to those queries incorporated that would be awesome. 

We can do another pass thru review to finalize this...

Thank you

**Comment by JustinCappos** [comment]
I'm assigning the issue to the reviewers now.  Please go through the document and make notes about any security issue, problems, questions, or concerns.   When you've finished, please say so in either here or the slack channel.  Once we get critical mass of reviewers, we will set up a meeting and discuss!

**Comment by JonZeolla** [comment]
My first review is complete

**Comment by JustinCappos** [comment]
The assessment has been completed!  Thanks to the reviewers and everyone from the Pixie side!
